lib/store/src/csf/testing-utils/index.ts

```diff
@@ -0,0 +1,102 @@
+  isExportStory,
+  AnyFramework,
+  AnnotatedStoryFn,
+  StoryAnnotations,
+  ComponentAnnotations,
+  Args,
+  StoryContext,
+} from '@storybook/csf';
+
+import { prepareStory } from '../prepareStory';
+import { normalizeStory } from '../normalizeStory';
+import { normalizeProjectAnnotations } from '../normalizeProjectAnnotations';
+import { HooksContext } from '../../hooks';
+import { NormalizedProjectAnnotations } from '../..';
+
+if (process.env.NODE_ENV === 'test') {
+  // eslint-disable-next-line global-require
+  const { default: addons, mockChannel } = require('@storybook/addons');
+  addons.setChannel(mockChannel());
+}
+
+export type StoryFile = {
+  default: Record<any, any>;
+  __esModule?: boolean;
+  __namedExportsOrder?: string[];
+};
+
+type PartialPlayFn = (
+  context: Partial<StoryContext> & Pick<StoryContext, 'canvasElement'>
+) => Promise<void> | void;
+
+export function composeStory<
+  TFramework extends AnyFramework = AnyFramework,
+  TArgs extends Args = Args
+>(
+  story: AnnotatedStoryFn<TFramework, TArgs> | StoryAnnotations<TFramework, TArgs>,
+  meta: ComponentAnnotations<TFramework, TArgs>,
+  globalConfig: NormalizedProjectAnnotations<TFramework> = {}
+) {
+  if (story === undefined) {
+    throw new Error('Expected a story but received undefined.');
+  }
+
+  const normalizedMeta = normalizeProjectAnnotations(meta);
+
+  const normalizedStory = normalizeStory(story.name, story, normalizedMeta);
+
+  const preparedStory = prepareStory<TFramework>(normalizedStory, normalizedMeta, globalConfig);
+
+  const defaultGlobals = Object.entries(globalConfig.globalTypes || {}).reduce(
+    (acc, [arg, { defaultValue }]) => {
+      if (defaultValue) {
+        acc[arg] = defaultValue;
+      }
+      return acc;
+    },
+    {} as Record<string, { defaultValue: any }>
+  );
+
+  const composedStory = (extraArgs: Partial<TArgs>) => {
+    const context: Partial<StoryContext> = {
+      ...preparedStory,
+      hooks: new HooksContext(),
+      globals: defaultGlobals,
+      args: { ...preparedStory.initialArgs, ...extraArgs },
+    };
+
+    return preparedStory.unboundStoryFn(context as StoryContext);
+  };
+
+  composedStory.storyName = story.storyName || story.name;
+  composedStory.args = preparedStory.initialArgs;
+  composedStory.play = preparedStory.playFunction as PartialPlayFn;
+  composedStory.parameters = preparedStory.parameters;
+
+  return composedStory;
+}
+
+export function composeStories<TModule extends StoryFile>(
+  storiesImport: TModule,
+  globalConfig: NormalizedProjectAnnotations<AnyFramework>,
+  composeStoryFn: typeof composeStory
+) {
+  const { default: meta, __esModule, __namedExportsOrder, ...stories } = storiesImport;
+  const composedStories = Object.entries(stories).reduce((storiesMap, [key, _story]) => {
+    if (!isExportStory(key as string, meta)) {
+      return storiesMap;
+    }
+
+    const storyName = String(key);
+    const story = _story as any;
+    story.storyName = storyName;
```

Comment:
Can you tell me what's happening here? What if `story.storyName` is already set?

Reply:
Would it make sense to use the normalised version of the story for this kind of logic? Then you can rest assured that `story.storyName` will always be there.

Reply:
I don't think so, as the normalised version of the story does not take the exports name into account in case no name was set to that story. The exports name is also lost by the time the story reaches `prepareStory`

Reply:
I checked what comes out of prepareStory and got this as a result:

```js
story: {
  // ... other properties
  componentId: 'csf3-accountform',
  title: 'CSF3/AccountForm',
  kind: 'CSF3/AccountForm',
  id: 'csf3-accountform',
  name: '',
  story: '',  
}
```

Reply:
Added support for exportsName in composeStory